ports/zephyr/zephyr_fs.c

```diff
@@ -0,0 +1,782 @@
+    }
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_3(zephyr_fs_rename_obj, zephyr_fs_rename);
+
+static mp_import_stat_t zephyr_fs_import_stat(void *vfs_in, const char *path) {
+    zephyr_fs_obj_t *self = vfs_in;
+    struct fs_dirent stats;
+    assert(self != NULL);
+    const char *ppath = zephyr_fs_make_path(self, mp_obj_new_str_from_cstr(path));
+    int err;
+
+    err = fs_stat(ppath, &stats);
+    if (err == 0) {
+        if (stats.type == FS_DIR_ENTRY_DIR) {
+            return MP_IMPORT_STAT_DIR;
+        } else {
+            return MP_IMPORT_STAT_FILE;
+        }
+    }
+    return MP_IMPORT_STAT_NO_EXIST;
+}
+
+static mp_obj_t zephyr_fs_stat(mp_obj_t vfs_in, mp_obj_t path_in) {
+    zephyr_fs_obj_t *self = MP_OBJ_TO_PTR(vfs_in);
+    int err;
+    struct fs_dirent stats;
+    const char *path = zephyr_fs_make_path(self, path_in);
+
+    err = fs_stat(path, &stats);
+    if (err < 0) {
+        mp_raise_msg_varg(&mp_type_OSError,
+            MP_ERROR_TEXT("error getting stats of Zephyr File System at %s: %q"), path, mp_errno_to_str(MP_OBJ_NEW_SMALL_INT(-err)));
+        return mp_const_none;
+    }
+
+    mp_obj_tuple_t *t = MP_OBJ_TO_PTR(mp_obj_new_tuple(10, NULL));
+    mp_int_t mode = 0;
+    if (stats.type == FS_DIR_ENTRY_DIR) {
+        mode |= MP_S_IFDIR;
+    } else {
+        mode |= MP_S_IFREG;
+    }
+    t->items[0] = MP_OBJ_NEW_SMALL_INT(mode);     // st_mode
+    t->items[1] = MP_OBJ_NEW_SMALL_INT(0);     // st_ino
+    t->items[2] = MP_OBJ_NEW_SMALL_INT(0);     // st_dev
+    t->items[3] = MP_OBJ_NEW_SMALL_INT(0);     // st_nlink
+    t->items[4] = MP_OBJ_NEW_SMALL_INT(0);     // st_uid
+    t->items[5] = MP_OBJ_NEW_SMALL_INT(0);     // st_gid
+    t->items[6] = mp_obj_new_int_from_uint(stats.size);     // st_size
+    t->items[7] = mp_obj_new_int_from_uint(0);     // st_atime
+    t->items[8] = mp_obj_new_int_from_uint(0);     // st_mtime
+    t->items[9] = mp_obj_new_int_from_uint(0);     // st_ctime
+
+    return MP_OBJ_FROM_PTR(t);
+}
+static MP_DEFINE_CONST_FUN_OBJ_2(zephyr_fs_stat_obj, zephyr_fs_stat);
+
+static mp_obj_t zephyr_fs_statvfs(mp_obj_t vfs_in, mp_obj_t path_in) {
+    zephyr_fs_obj_t *self = MP_OBJ_TO_PTR(vfs_in);
+    (void)path_in;
+    int err;
+    struct fs_statvfs stats;
+
+    err = fs_statvfs(self->mount->mnt_point, &stats);
+    if (err < 0) {
+        mp_raise_msg_varg(&mp_type_OSError,
+            MP_ERROR_TEXT("error getting stats of Zephyr File System: %q"), mp_errno_to_str(MP_OBJ_NEW_SMALL_INT(-err)));
+        return mp_const_none;
+    }
+
+    mp_obj_tuple_t *t = MP_OBJ_TO_PTR(mp_obj_new_tuple(10, NULL));
+
+    t->items[0] = MP_OBJ_NEW_SMALL_INT(stats.f_bsize);     // f_bsize
+    t->items[1] = MP_OBJ_NEW_SMALL_INT(stats.f_frsize);     // f_frsize
+    t->items[2] = MP_OBJ_NEW_SMALL_INT(stats.f_blocks);     // f_blocks
+    t->items[3] = MP_OBJ_NEW_SMALL_INT(stats.f_bfree);     // f_bfree
+    t->items[4] = t->items[3];     // f_bavail
```

Comment:
I don't think these values are correct.  If I do `mpremote df` I get:
```
$ mpremote df
filesystem     size     used    avail use% mounted on
<FS>          90112      512    89600    0 /flash
```
But the same filesystem with `FlashArea` is:
```
$ mpremote df
filesystem     size     used    avail use% mounted on
<VfsLfs2>   1441792     8192  1433600    0 /flash
```
So the FS implementation here is a factor of 16 too small.

Reply:
What platform is that on? RP2040? It's very possible this is coming from different settings so would need to be compared to the settings of the DTS instead

Reply:
well added a if, but dont know what is happening on zephry side... done

Reply:
> What platform is that on? RP2040?

Yes, sorry, should have mentioned it was a standard Pico with 2MiB flash, using the `rpi_pico` zephyr board definition from this repo.